### PR TITLE
Add support for "fractional gates"

### LINKED ISSUE #413: Ensure support for new `rx` and `rzz` gates

### PR DESCRIPTION
**Pull request by cqc-alec**
Closes #413 .
The test failure is unrelated: see #449 .
It is not currently possible to add a CI test for this as none of the devices on the open plan support fractional gates. But I have tested locally with `ibm_torino`.
In addition, I have added a squashing step to the default compilation passes, since they were not squashing single-qubit gates (as pointed out in the comments on the issue).

### CODE REVIEW

File: pytket/extensions/qiskit/backends/ibm.py [hunk 1]
```diff
@@ -128,13 +129,16 @@ def _save_ibmq_auth(qiskit_config: Optional[QiskitConfig]) -> None:
...
+
+
 def _get_primitive_gates(gateset: set[OpType]) -> set[OpType]:
-    if gateset >= {OpType.X, OpType.SX, OpType.Rz, OpType.CX}:
-        return {OpType.X, OpType.SX, OpType.Rz, OpType.CX}
-    elif gateset >= {OpType.X, OpType.SX, OpType.Rz, OpType.ECR}:
-        return {OpType.X, OpType.SX, OpType.Rz, OpType.ECR}
-    else:
-        return gateset
+    return gateset & (ALL_PRIMITIVE_1Q_GATES | ALL_PRIMITIVE_2Q_GATES)
```
[the remark this answers is not in the document]
Comment (cqc-alec) [reply pytket/extensions/qiskit/backends/ibm.py:137]: I don't think this should be a problem. In fact it looks like none of the devices have `CX` any more; they use `CZ` instead, though never in combination with `ECR`. With fractional gates some devices have both `CZ` and `ZZPhase`. This doesn't seem to cause any issues.

**Review by cqc-melf — APPROVED** [review]
